test/e2e/metrics_test.go

```diff
@@ -86,36 +101,63 @@ func NewMetricsTestConfig(t *testing.T, client, selector, clusterRole, clusterBi
 	}
 }
 
-// run will execute all steps of those tests
+// run will execute all steps of those tests with retries.
 func (c *MetricsTestConfig) run() {
-	c.createMetricsClusterRoleBinding()
-	token := c.getServiceAccountToken()
-	c.createCurlMetricsPod()
-	c.validate(token)
+	err := retry(5, 5*time.Second, func() error {
```

Comment:
I think the retry stuff is way more than what we need for  test. Is it not working without the retry logic?

Reply:
it's gone now